src/NodeVisitor/ClassUseVisitor.php

```diff
@@ -0,0 +1,111 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Nusje2000\ComposerMonolith\NodeVisitor;
+
+use LogicException;
+use Nusje2000\ComposerMonolith\Helper\PackageFinderInterface;
+use Nusje2000\ComposerMonolith\Validator\Violation\OutOfBoundsReferenceViolation;
+use Nusje2000\ComposerMonolith\Validator\ViolationCollection;
+use Nusje2000\DependencyGraph\DependencyGraph;
+use PhpParser\Node;
+use PhpParser\NodeVisitorAbstract;
+
+final class ClassUseVisitor extends NodeVisitorAbstract
+{
+    /**
+     * @var string
+     */
+    private $fileName;
+
+    /**
+     * @var DependencyGraph
+     */
+    private $graph;
+
+    /**
+     * @var PackageFinderInterface
+     */
+    private $packageFinder;
+
+    /**
+     * @var ViolationCollection
+     */
+    private $violations;
+
+    public function __construct(DependencyGraph $graph, PackageFinderInterface $packageFinder)
+    {
+        $this->packageFinder = $packageFinder;
+        $this->graph = $graph;
+        $this->violations = new ViolationCollection();
+    }
+
+    public function enterNode(Node $node): ?int
+    {
+        if (!$node instanceof Node\Name) {
+            return null;
```

Comment:
Doesn't this imply incorrect usage of this method? If so, I would throw an exception.